templates/templates/base.html

```diff
@@ -78,6 +78,21 @@
   {% if not(hide_nav == True) %}
     {% include "templates/_navigation.html" %}
   {% endif %}
+  <div class="p-strip is-shallow u-no-padding--bottom" id="success">
+    <div class="row">
+      <div class="p-notification--positive">
+        <p class="p-notification__response">Submittion was successful.</p>
```

Comment:
This is fairly neat, but are you sure this is better than sending the user to a default thank-you page? A thank-you page is a bit more straightforward and obvious.

Reply:
ReturnURL field is still respected. But this is a fallback for forms without a returnURL set.

Reply:
I understand that, I'm asking about the fall back.

Reply:
It's just that your implementation is a little brittle for a couple of reasons:

- you're redirecting them to the "referer" header, which *may* not exist for any number of reasons, including that their browser may deliberately block it.
- changing the hash value will remove the notification - e.g. if someone expands the top nav. This could be confusing.
- even if the notification displays fine, it could still be confusing to end up on a page that has nothing to do with the form you just submitted

So you go to "/blog", you see the newsletter form, you sign up for the newsletter. I find it a little hard to decide whether the user is expecting to be sent back to /blog. I think they probably don't, they probably want to be sure their sign up was successful.

It just seems to me that the more straightforward solution - both technically and for the user's experience - is to send them to a dedicated URL that says "thank you for submitting the form". There's no ambiguity; you're not trying to do two different things with a single page which is usually a bad idea.

If we have a referer value, we could put a link on that page saying "go back where you came from", or if you feel strongly we could automatically redirect the user back after 5 seconds or something (although I'm not crazy about that either).

And if this is just the fallback anyway (as in, well crafted forms will have a better target anyway), then I'd say we should go for the simpler solution that's less likely to break.

Reply:
Your right this is brittle. I had removed the referrer and landed the user on a new generic thank you page.